bin/fastq_dir_to_samplesheet.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import glob
+import argparse
+
+def parse_args(args=None):
+    Description = "Generate nf-core/rnaseq samplesheet from a directory of FastQ files."
+    Epilog = "Example usage: python fastq_dir_to_samplesheet.py <FASTQ_DIR> <SAMPLESHEET_FILE>"
+
+    parser = argparse.ArgumentParser(description=Description, epilog=Epilog)
+    parser.add_argument("FASTQ_DIR", help="Folder containing raw FastQ files.")
+    parser.add_argument("SAMPLESHEET_FILE", help="Output samplesheet file.")
+    parser.add_argument('-st', '--strandedness', type=str, dest="STRANDEDNESS", default='unstranded', help="Value for 'strandedness' in samplesheet. Must be one of 'unstranded', 'forward', 'reverse'.")
+    parser.add_argument('-r1', '--read1_extension', type=str, dest="READ1_EXTENSION", default='_R1_001.fastq.gz', help="File extension for read 1.")
+    parser.add_argument('-r2', '--read2_extension', type=str, dest="READ2_EXTENSION", default='_R2_001.fastq.gz', help="File extension for read 2.")
+    parser.add_argument('-se', '--single_end', dest="SINGLE_END", action='store_true', help="Single-end information will be auto-detected but this option forces paired-end FastQ files to be treated as single-end so only read 1 information is included in the samplesheet.")
+    parser.add_argument('-sn', '--sanitise_name', dest="SANITISE_NAME", action='store_true', help="Whether to further sanitise FastQ file name to get sample id. Used in conjunction with --sanitise_name_delimiter and --sanitise_name_index.")
+    parser.add_argument('-sd', '--sanitise_name_delimiter', type=str, dest="SANITISE_NAME_DELIMITER", default='_', help="Delimiter to use to sanitise sample name.")
+    parser.add_argument('-si', '--sanitise_name_index', type=int, dest="SANITISE_NAME_INDEX", default=1, help="After splitting FastQ file name by --sanitise_name_delimiter all elements before this index (1-based) will be joined to create final sample name.")
+    return parser.parse_args(args)
+
+def fastq_dir_to_samplesheet(fastq_dir, samplesheet_file, strandedness='unstranded', read1_extension='_R1_001.fastq.gz', read2_extension='_R2_001.fastq.gz', single_end=False, sanitise_name=False, sanitise_name_delimiter='_', sanitise_name_index=1):
+
+    ## Get read 1 files
+    read_dict = {}
+    read1_files = glob.glob(os.path.join(fastq_dir, f'*{read1_extension}'), recursive=False)
+    for read1_file in read1_files:
+        sample = os.path.basename(read1_file).replace(read1_extension,'')
+        if sanitise_name:
+            sample = sanitise_name_delimiter.join(os.path.basename(read1_file).split(sanitise_name_delimiter)[:sanitise_name_index])
+        read_dict[sample] = [read1_file]
```

Comment:
Does this still work if samples have been re-sequenced (to be `cat`ed by the pipeline)?

Reply:
Yes, it should do but you would need to use the sanitise parameters in the script to get just the main id of the pipeline and strip out all of the lane info etc.

Reply:
Hang on, but the dict key is the sample_id, so there can only be one pair of fastq files per sample id. Resequenced samples would require two lines in the samplesheet with the same sample_id. Am I missing something?

Reply:
You are absolutely right 🤦🏽 I initially wrote this script for viralrecon where merging isn't really required but I should have tested a little more thoroughly. 

Fixed https://github.com/nf-core/rnaseq/pull/660/commits/17a15fe233e3f9d6a53744dc4ed45b43a910cdb9

Given the directory structure below:

```console
$ mkdir fastq
$ cd fastq
$ touch DAN2501A41_S3_L001_R1_001.fastq.gz
$ touch DAN2501A41_S3_L001_R2_001.fastq.gz
$ touch DAN2501A41_S3_L002_R1_001.fastq.gz
$ touch DAN2501A41_S3_L002_R2_001.fastq.gz
$ touch DAN2501A42_S4_L001_R1_001.fastq.gz
$ touch DAN2501A42_S4_L001_R2_001.fastq.gz
$ touch DAN2501A42_S4_L002_R1_001.fastq.gz
$ touch DAN2501A42_S4_L002_R2_001.fastq.gz
$ touch DAN2501A43_S5_L001_R1_001.fastq.gz
$ touch DAN2501A43_S5_L001_R2_001.fastq.gz
```

Running the command below will add separate entries for each library:

```console
./fastq_dir_to_samplesheet.py fastq test.csv --sanitise_name
```

Reply:
LGTM now!